### ISSUE TITLE
Issue with 2.4.2-p2 and vertex/

### ISSUE THREAD

**Issue by JelleGe** [yireo/magento2-replace-all#22]
Seems that it's not allowed to remove the vertex/* modules in 2.4.2-p2.

Can you confirm this?

**Comment by jissereitsma** [comment]
I'm not sure if *allowing* here is a thing. I'm allowing it, composer is allowing it and I think Vertex actually does not know what composer replacements are about. Could it be that you are trying to address another issue instead? What makes you believe that permissions are a problem here?

**Comment by JelleGe** [comment]
Thanks for the replay.

We face in that case the following error:

```
  Problem 1
    - magento/project-community-edition is present at version 2.4.1 and cannot be modified by Composer
    - magento/product-community-edition 2.4.2-p2 requires vertexinc/product-magento-module 4.2.0 -> satisfiable by vertexinc/product-magento-module[4.2.0].
    - vertexinc/product-magento-module[4.2.0] cannot be installed as that would require removing magento/project-community-edition[2.4.1]. They both replace vertex/product-magento-module and thus cannot coexist.
    - Root composer.json requires magento/product-community-edition 2.4.2-p2 -> satisfiable by magento/product-community-edition[2.4.2-p2].
```

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

**Comment by jissereitsma** [comment]
Ok, this clearifies things. My recommendation for opening up any issue in any repository would be to not be vague, but copy the actual error message. *Magento allowing something* is totally different from the actual composer error that you posted. Posting the error saves us both time.

As you can see, the error involves a module `vertexinc/product-magento-module` which is already replaced in the `magento2-replace-bundled` repo: https://github.com/yireo/magento2-replace-bundled/blob/magento-2.4/composer.json#L40 But the `magento2-replace-all` repo seemed to not have included this fix yet. I have now released 4.2.3 (https://github.com/yireo/magento2-replace-all/releases/tag/4.2.3) to include this fix as well here.

**Comment by JelleGe** [comment]
Completely agree, my bad! Thanks for this fix and fast response.
